lib/sync.js

```diff
@@ -742,6 +753,12 @@ SyncApi.prototype._processSyncResponse = function(syncToken, data) {
                 room.currentState.paginationToken = syncToken;
                 self._deregisterStateListeners(room);
                 room.resetLiveTimeline(joinObj.timeline.prev_batch);
+
+                // XXX: for now we have to assume any gap in any timeline is
```

Comment:
This doesn't seem like a thing that is likely to change, as long as you are populating the notification timeline from the room timelines?